sklearn/model_selection/_validation.py

```diff
@@ -474,6 +476,18 @@ def _fit_and_predict(estimator, X, y, train, test, verbose, fit_params,
         estimator.fit(X_train, y_train, **fit_params)
     func = getattr(estimator, method)
     predictions = func(X_test)
+    if method in ['decision_function', 'predict_proba', 'predict_log_proba']:
```

Comment:
The problem lies is that `predict_proba` and `predict_log_proba` return a 2D array when there are 2 classes in the training set but `decision_function` returns a 1D array. So, I chose to go by method to differentiate the two.